spotbugs/src/main/java/edu/umd/cs/findbugs/sarif/SarifBugReporter.java

```diff
@@ -59,27 +74,49 @@ private void processInvocations(JSONWriter jsonWriter, @NonNull Map<URI, String>
         Invocation invocation = new Invocation(exitCode,
                 getSignalName(exitCode), exitCode == 0,
                 execNotifications, configNotifications);
-        jsonWriter.key("invocations").array().value(invocation.toJSONObject());
-        jsonWriter.endArray();
+        try {
+            Gson gson = new Gson();
+            jsonWriter.name("invocations").beginArray();
+            gson.toJson(invocation.toJsonObject(), jsonWriter);
+            jsonWriter.endArray();
+        } catch (IOException e) {
+            throw e;
+        }
     }
 
-    private void processTool(@NonNull JSONWriter jsonWriter, @NonNull JSONArray rules) {
-        jsonWriter.key("tool").object();
-        processExtensions(jsonWriter);
-        jsonWriter.key("driver").object();
-        jsonWriter.key("name").value("SpotBugs");
-        // Eclipse plugin does not follow the semantic-versioning, so use "version" instead of "semanticVersion".
-        jsonWriter.key("version").value(Version.VERSION_STRING);
-        // SpotBugs refers JVM config to decide which language we use.
-        jsonWriter.key("language").value(Locale.getDefault().getLanguage());
-        jsonWriter.key("rules").value(rules);
-        jsonWriter.endObject().endObject();
+    private void processTool(@NonNull JsonWriter jsonWriter, @NonNull JsonArray rules) throws IOException {
+        try {
+            Gson gson = new Gson();
```

Comment:
Is it OK to generate `Gson` instance on demand? Not sure its cost, but if it takes much resources (CPU or Java heap), consider to create it once and reuse by providing it via method parameter.